### PR TITLE
feat(console): show app secret

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Show app secret in AC for traditional app.

## Testing

截屏2022-08-02 下午5 23 24

截屏2022-08-02 下午5 23 28

### CODE REVIEW

**Comment by demonzoo** [comment]
The "Eye" icon should be closed when hiding the secret

File: packages/console/src/components/CopyToClipboard/index.tsx [hunk 1]
```diff
@@ -41,7 +56,14 @@ const CopyToClipboard = ({ value, className, variant = 'contained' }: Props) =>
...
       <div className={styles.row}>
-        {variant === 'icon' ? null : value}
+        {variant === 'icon' ? null : <div className={styles.content}>{valueToShow}</div>}
+        {isContentHidden && (
+          <div className={styles.eye}>
+            <IconButton onClick={toggleHiddenContent}>
+              <Eye />
+            </IconButton>
+          </div>
+        )}
```
Comment (demonzoo) [inline packages/console/src/components/CopyToClipboard/index.tsx:66]: Should have a "closed eye" icon when hiding the secret

File: packages/console/src/components/CopyToClipboard/index.tsx [hunk 2]
```diff
@@ -12,14 +13,24 @@ type Props = {
...
   const [copyState, setCopyState] = useState<CopyState>('copy');
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console.general' });
+  const [showHiddenContent, setShowHiddenContent] = useState(false);
+
+  const valueToShow = useMemo(() => {
+    if (!isContentHidden || showHiddenContent) {
+      return value;
+    }
+
+    return '*'.repeat(value.length);
```
Comment (darcyYe) [inline packages/console/src/components/CopyToClipboard/index.tsx:33]: nit: it seems the length of the mask doesn't need to be the same length as the origin values.

**Comment by wangsijie** [comment]
update: add eye close icon

image

File: packages/console/src/components/CopyToClipboard/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/components/CopyToClipboard/index.tsx]: updated, PTAL

File: packages/console/src/pages/ApplicationDetails/components/Settings.tsx [hunk 3]
```diff
@@ -64,6 +64,16 @@ const Settings = ({ applicationType, oidcConfig, defaultData, isDeleted }: Props
           placeholder={t('application_details.description_placeholder')}
         />
       </FormField>
+      {applicationType === ApplicationType.Traditional && (
+        <FormField title="application_details.application_secret" className={styles.textField}>
+          <CopyToClipboard
+            isContentHidden
```
Comment (darcyYe) [inline packages/console/src/pages/ApplicationDetails/components/Settings.tsx:70]: Do we have a default value to determine whether contents in some CopyToClipboard are displayed by default?
I cannot find another appearance of `isContentHidden` in this file to determine its value.

File: packages/console/src/components/CopyToClipboard/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/components/CopyToClipboard/index.tsx:33]: But it's not wrong to be the same?

File: packages/console/src/pages/ApplicationDetails/components/Settings.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/pages/ApplicationDetails/components/Settings.tsx:70]: `isContentHidden` is enough, its default value is false.

File: packages/console/src/components/CopyToClipboard/index.tsx (hunk 1 above)
Comment (demonzoo) [reply packages/console/src/components/CopyToClipboard/index.tsx]: Should update the screenshot in PR description as well. Cheers

**Review by darcyYe — APPROVED** [review]
LGTM

File: packages/console/src/components/CopyToClipboard/index.tsx [hunk 4]
```diff
@@ -12,14 +14,24 @@ type Props = {
...
```
Comment (demonzoo) [inline packages/console/src/components/CopyToClipboard/index.tsx:33]: Will the length always be 32? Just asking

Comment (wangsijie) [reply packages/console/src/components/CopyToClipboard/index.tsx:33]: 64

File: packages/console/src/components/CopyToClipboard/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/components/CopyToClipboard/index.tsx]: ha, I add a comment to keep the modify history.

File: packages/console/src/components/CopyToClipboard/index.tsx [hunk 5]
```diff
@@ -12,14 +14,24 @@ type Props = {
   value: string;
   className?: string;
   variant?: 'text' | 'contained' | 'border' | 'icon';
+  isContentHidden?: boolean;
```
Comment (demonzoo) [inline packages/console/src/components/CopyToClipboard/index.tsx:17]: `isContentHidden` is actually the flag to tell the component if the content visibility can be toggled, not to hide the content... So it's a bit confusing IMO.
Any better name? How about `hasVisibilityToggle`?

File: packages/console/src/components/CopyToClipboard/index.tsx [hunk 6]
```diff
@@ -41,7 +57,14 @@ const CopyToClipboard = ({ value, className, variant = 'contained' }: Props) =>
       }}
     >
       <div className={styles.row}>
-        {variant === 'icon' ? null : value}
+        {variant === 'icon' ? null : <div className={styles.content}>{valueToShow}</div>}
```
Comment (demonzoo) [inline packages/console/src/components/CopyToClipboard/index.tsx:60]:
```suggestion
        {variant !== 'icon' && <div className={styles.content}>{valueToShow}</div>}
```

File: packages/console/src/components/CopyToClipboard/index.tsx [hunk 7]
```diff
@@ -12,14 +14,24 @@ type Props = {
...
 type CopyState = TFuncKey<'translation', 'admin_console.general'>;
 
-const CopyToClipboard = ({ value, className, variant = 'contained' }: Props) => {
+const CopyToClipboard = ({ value, className, isContentHidden, variant = 'contained' }: Props) => {
   const copyIconReference = useRef<HTMLDivElement>(null);
   const [copyState, setCopyState] = useState<CopyState>('copy');
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console.general' });
+  const [showHiddenContent, setShowHiddenContent] = useState(false);
+
+  const valueToShow = useMemo(() => {
```
Comment (demonzoo) [inline packages/console/src/components/CopyToClipboard/index.tsx:28]: `valueToShow` => `displayValue`

File: packages/console/src/icons/EyeClose.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { SVGProps } from 'react';
+
+const EyeClose = (props: SVGProps<SVGSVGElement>) => (
```
Comment (demonzoo) [inline packages/console/src/icons/EyeClose.tsx:3]: nit: `EyeClosed`

File: packages/console/src/components/CopyToClipboard/index.tsx (hunk 5 above)
Comment (wangsijie) [reply packages/console/src/components/CopyToClipboard/index.tsx]: updated

File: packages/console/src/components/CopyToClipboard/index.tsx (hunk 7 above)
Comment (wangsijie) [reply packages/console/src/components/CopyToClipboard/index.tsx]: updated

File: packages/console/src/components/CopyToClipboard/index.tsx [hunk 9]
```diff
@@ -41,7 +57,14 @@ const CopyToClipboard = ({ value, className, variant = 'contained' }: Props) =>
       }}
     >
       <div className={styles.row}>
-        {variant === 'icon' ? null : value}
+        {variant === 'icon' ? null : <div className={styles.content}>{valueToShow}</div>}
+        {isContentHidden && (
+          <div className={styles.eye}>
+            <IconButton onClick={toggleHiddenContent}>
+              {showHiddenContent ? <EyeClose /> : <Eye />}
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/console/src/components/CopyToClipboard/index.tsx]: updated

File: packages/console/src/icons/EyeClose.tsx (hunk 8 above)
Comment (wangsijie) [reply packages/console/src/icons/EyeClose.tsx]: updated

**Pull request merged**